src/App.js

```diff
@@ -1,16 +1,55 @@
-import React from 'react';
+import React, { useState } from 'react';
 import './App.css';
 import chatMessages from './data/messages.json';
+import ChatLog from './components/ChatLog';
 
 const App = () => {
+  const [entriesData, setEntriesData] = useState(chatMessages);
+
+  const calcLikes = (entriesData) => {
+    let likeCount = 0;
+
+    for (const entry of entriesData) {
+      if (entry.liked) {
+        likeCount++;
+      }
+    }
+    return likeCount;
+  };
+
+  const getParticipants = (entriesData) => {
+    let chatMembers = new Set();
+
+    for (const entry of entriesData) {
+      chatMembers.add(entry.sender);
+    }
+
+    return Array.from(chatMembers).join(' and ');
+  };
```

Comment:
Nice, we could also just hardcode the senders with entriesData[0].sender and entriesData[1].sender, but this adds flexibility and could be reused if the chat has more than 2 participants